src/util/sharing_map.h

```diff
@@ -128,6 +134,7 @@ class sharing_mapt
   friend void sharing_map_copy_test();
   friend void sharing_map_collision_test();
   friend void sharing_map_view_test();
+  friend void sharing_map_sharing_stats_test();
```

Comment:
At the risk of getting side-tracked: couldn't those tests be implemented by deriving a new child that provides more access? Those far-away friends are a bit of a pain point...

Reply:
Do you mean deriving a new class from `sharing_mapt` in the unit tests which has public methods that wrap the protected members that we want to test? Alternatively, we could also derive a new class from `sharing_mapt` in the unit tests which then includes those friend declarations above.

What is the issue with the current approach though? Since the friends declared above are only from the unit tests it seems relatively safe.

Reply:
Far-away friends are easy to lose touch with. In code terms, I effectively need to `git grep` to figure out who you are trying to be friends with. Maybe the solution is just adding a comment :-)  In practice, adding a new test currently requires touching this code, which really shouldn't be necessary. Hence my preference for deriving a new class from `sharing_mapt` in the unit tests, which would restrict any changes in unit tests to the unit test code.

Reply:
Ah, I see. Yeah, I'll derive from `sharing_mapt` in the unit tests.